FD-3145: Transactions List

albatahmet opened the pull request.

albatahmet committed:
main layout, card komponent

albatahmet committed:
Transactions Pages - Transaction Detail

Eksikler
Bütün gelen veride zaman filtreleme çalışmadı
footer componenti için lazy konuşulacak
detail kısmındaki API çalışıyor fakat response data boş dönüyor
!!! Dökümantasyonda gösterilen api responsları ile şuanda gelen API responsları aynı değil !!! veriler eksik geliyor Hepsi çok önemli veriler!!!!

k-cem-fd commented on components/Menu.js:
```diff
@@ -1,7 +1,7 @@
 
-export function Menu({ open, setMenuOpen }) {
-  const [selectedMenuItem, setSelectedMenuItem] = useState("Home");
+export function Menu({ open, setMenuOpen,selected="" }) {
```
You should use ESlint @albatahmet

k-cem-fd commented on components/Menu.js:
```diff
@@ -44,7 +44,7 @@ export function Menu({ open, setMenuOpen }) {
           {MOBILE_MENU_ITEMS.map((m, i) => {
             return (
-              <div key={`menu-item-${i}`} className="flex justify-center w-1/4">
+              <div key={`menu-item-${i}`} className="flex justify-center w-1/4" onClick={()=> m?.href ? window.location.href = `/${m?.href}`:"" }>
```
When you change it to lazy, this will be deleted. But if you want to redirect user to new path again please create a function and add this code block =>  "router.push('/personal-transactions');"
In general perspective, Next.js is giving us routing capability, why we bother to use href 😉

k-cem-fd commented on components/Menu.js:
```diff
@@ -1,7 +1,7 @@
-export function Menu({ open, setMenuOpen }) {
-  const [selectedMenuItem, setSelectedMenuItem] = useState("Home");
+export function Menu({ open, setMenuOpen,selected="" }) {
+  const [selectedMenuItem, setSelectedMenuItem] = useState(selected) || "Home";
```
Please use meaningful variables, selected what ? 🤔

k-cem-fd commented on components/TransactionsPage/index.js:
```diff
@@ -0,0 +1,57 @@
+import React, { useEffect, useState } from 'react';
```
Why did you write whole component into index.js ? Please create new file TransactionsPage and move code.
In index.js file you should make only export as below to import easily in another file.
export { TransactionsPage } from './TransactionsPage';

k-cem-fd commented on components/TransactionsPage/index.js:
```diff
@@ -0,0 +1,57 @@
+ 
+
+  useEffect(() => {
+    console.log('test', new Date('2022-09-07T00:00:00Z') - new Date('2022-09-05T00:00:00Z'));
```
Please remove all console.log() which you added.

k-cem-fd commented on components/TransactionsPage/index.js:
```diff
@@ -0,0 +1,57 @@
+
+export function Transaction({ data = [], loading = '' }) {
+  const [transaction, setTransaction] = useState([]);
+  const [isOpen, setisOpen] = useState(false);
```
Did you use 'isOpen' anywhere in the file ? If no, please remove. And it is not meaningful also, what is open 🤔

k-cem-fd commented on components/TransactionsPage/transaction-card.js:
```diff
@@ -0,0 +1,34 @@
+  }
+    const formatter = (date) =>{
+      return date.slice(11,16)
+   }
```
postDate value is like this -> "2022-09-06T00:00:00Z" Why do you get slice(11, 16), you should slice(0, 10)

![image](https://user-images.githubusercontent.com/97659632/188594685-19f00c20-f416-42a6-b525-a4ba912cfe28.png)

k-cem-fd commented on components/TransactionsPage/transactions-info.js:
```diff
@@ -0,0 +1,112 @@
+  .get(`/api/transaction-info`,{params : {userId, transactionId}})
+  .then(function (response) {
+    console.log(response);
+    // setData(response.data)
```
Please remove commented code

k-cem-fd commented on components/TransactionsPage/transactions-info.js:
```diff
@@ -0,0 +1,112 @@
+   		stringData = stringData.replaceAll('T', ' ')
+   		stringData = stringData.replaceAll('Z', '')
+   		stringData = stringData.replaceAll('+03:00', '')
+   		return stringData
```
What are you doing with this logic 🤔

k-cem-fd commented on components/TransactionsPage/transactions-info.js:
```diff
@@ -0,0 +1,112 @@
+
+   useEffect(() => {
+     if(isOpen) getInfo()
+   }, [isOpen])
```
Is this working as expected, it crashed on my local

k-cem-fd commented on pages/api/transactions.js:
```diff
@@ -0,0 +1,27 @@
+const axios = require('axios');
+const { getBasiqAuthorizationHeader } = require('../../serverAuthentication');
+
```
Please remove empty lines

k-cem-fd commented on pages/personal-transactions.js:
```diff
@@ -0,0 +1,44 @@
+import { useToggleState } from '../utils/useToggleState';
+
+export default function PersonalFinance() {
+  const [data, setData] = useState([])
```
When you implement lazy load, this file will be deleted

k-cem-fd commented on yarn.lock:
```diff
@@ -2,75 +2,46 @@
 # yarn lockfile v1
 
 
-"@babel/code-frame@7.12.11":
```
Please don't send your local yarn.lock file

albatahmet committed:
React Lazy

React Lazy component has been integrated. css fixes are needed to work properly with other fields

albatahmet committed:
test Fix work

test Fix work

k-cem-fd committed:
FD-3145: .env.sample file is added

k-cem-fd committed:
FD-3145: eslint conf is reverted

k-cem-fd committed:
FD-3145: gitignore is reverted

k-cem-fd committed:
FD-3145: gitignore is reverted

k-cem-fd committed:
FD-3145: Main merge

k-cem-fd committed:
FD-3145: package.json is reverted

k-cem-fd committed:
FD-3145: yarn.lock is reverted

k-cem-fd committed:
FD-3145: PersonalFinance component is reverted

k-cem-fd committed:
FD-3145: Transactions and detail pages are updated

k-cem-fd committed:
FD-3145: Unique key error fix

k-cem-fd committed:
FD-3145: Empty lines removed
